src/main/java/it/gov/pagopa/admissibility/service/build/BeneficiaryRule2DroolsRuleImpl.java

```diff
@@ -93,4 +113,12 @@ private ExtraFilter automatedCriteria2ExtraFilter(AutomatedCriteriaDTO automated
         FilterOperator operator = automatedCriteriaDTO.getOperator();
         return new NotOperation(new Filter(field, operator, automatedCriteriaDTO.getValue()));
     }
+
+    private List<String> getAutomatedCriteriaCodes(List<AutomatedCriteriaDTO> initiativeAutomatedCriteria) {
+        List<String> automatedCriteriaCodes = new ArrayList<>();
+        for(AutomatedCriteriaDTO criterium : initiativeAutomatedCriteria) {
+            automatedCriteriaCodes.add(criterium.getCode());
+        }
+        return automatedCriteriaCodes;
```

Comment:
it's more efficient to use java8's streamApi or define the initialSize of the ArrayList inside the constructor, in order to avoid the inner array re-allocation when adding data